tests/test_control.py

```
@@ -448,6 +448,359 @@ def test_control_plane_bootstrap_uses_api_returned_mqtt_credentials():
+        "active": False,
+        "marketplace_skill_id": "marketplace-1",
+        "source_ref": "https://github.com/acme/skill-lobby",
+        "version_pin": "v1.2.0",
+    }
+
+
+def test_control_plane_surfaces_the_paid_plan_gate_as_an_api_error():
+    session = ProvisioningSession(error=(402, {"detail": "API access requires a paid plan."}))
+    api = provisioning_api(session)
+
+    with pytest.raises(ThalovantAPIError, match="HTTP 402: API access requires a paid plan."):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)